app/views/projects/_issue_summary.html.erb

```diff
@@ -40,106 +40,5 @@
-
-var x = d3.scale.ordinal()
-    .rangeRoundBands([0, width], .1);
-
-var y = d3.scale.linear()
-    .range([height, 0]);
-
-var xAxis = d3.svg.axis()
-    .scale(x)
-    .tickSize(0)
-    .orient("bottom");
-
-var yAxis = d3.svg.axis()
-    .scale(y)
-    .orient("left")
-    .tickSize(1, 1)
-    .tickFormat(d3.format(".0f"))
-    .ticks(2);
-    // .tickValues([1, 3, 4]);
-
-var svg = d3.select("#issue-chart").append("svg")
-    .attr("width", width + margin.left + margin.right)
-    .attr("height", height + margin.top + margin.bottom)
-  .append("g")
-    .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
-
-<%# --------------------------------------------------------- Ruby variables %>
-<% highest = 0 %>
-var data = [
-<% @tags.each do |tag| %>
-  <% list = @issues_by_tag[tag.name] %>
-  <% highest = list.count if list.count > highest %>
-  {letter: '<%= j(tag.display_name) %>', frequency: <%= list.count %>},
-<% end %>
-  {letter: 'N/A', frequency: <%= @issues_by_tag[:unassigned].count %>}
-];
-
-var x_domain = [
-  <% @tags.each do |tag| %>
-  '<%= j tag.display_name %>',
-  <% end %>
-  'N/A'
-];
-
-var highest_y = <%= [highest, @issues_by_tag[:unassigned].count].max %>;
-<%# -------------------------------------------------------- /Ruby variables %>
-
-
-// x.domain(['High', 'Medium', 'Low']);
-x.domain(x_domain);
-
-// y.domain([0, 5]);
-y.domain([0, highest_y]);
-
-svg.append("g")
-    .attr("class", "x axis")
-    .attr("transform", "translate(0," + height + ")")
-    .call(xAxis);
-
-// svg.append("g")
-//     .attr("class", "y axis")
-//     .call(yAxis)
-
-var bars = svg.append("g");
-
-bars.selectAll("rect")
-    .data(data)
-  .enter().append("rect")
-    // .attr("class", function(d){ return "bar-" + d.letter; } )
-    .attr("class", "bar" )
-    .attr("x", function(d) { return x(d.letter); })
-    .attr("width", x.rangeBand())
-    .attr("y", function(d) { return y(d.frequency); })
-    .attr("height", function(d) { return height - y(d.frequency); });
-
-
-bars.selectAll("text")
-    .data(data)
-  .enter().append("text")
-    .attr("x", function(d, i) { return x(d.letter) + x.rangeBand()/2; })
-    .attr("y", function(d) { return y(d.frequency);})
-    .attr("dy", -5)
-    .attr("text-anchor", "middle")
-    .attr("class", "counter" )
-    .text(function(d) {return d.frequency;});
-
-<% @tags.each_with_index do |tag, i| %>
-$($('.tick')[<%= i %>]).attr('fill', '<%=j tag.color %>');
-$($('.bar')[<%= i %>]).attr('fill', '<%=j tag.color %>');
-$($('.counter')[<%= i %>]).attr('fill', '<%=j tag.color %>');
-<% end %>
-$($('.tick')[<%= @tags.count %>]).attr('fill', '#ccc');
-$($('.bar')[<%= @tags.count %>]).attr('fill', '#ccc');
-$($('.counter')[<%= @tags.count %>]).attr('fill', '#ccc');
-</script>
+  <div id='issues-summary-data' data-tags='<%= @tags.map{ |t| [t.name, [t.display_name, t.color]] }.to_h.to_json %>' data-issues-count='<%= @count_by_tag.to_json %>'></div>
```

Comment:
If we're extracting the stats calculation up to the Controller for `@count_by_tag` wouldn't it make sense to do the same for the `@tags` and extract it too?